stream-chat-swift-docs/docs/guides/working-with-a-channel.md

```diff
@@ -1,3 +1,298 @@
 ---
 title: Working with a Channel
----
\ No newline at end of file
+---
+
+## Getting Channel Data
+
+Here's how you can get data of a Channel you already know the Channel ID of:
+
+```swift
+let channelController = chatClient.channelController(for: ChannelId(type: .messaging, id: <#Your Channel ID#>)
+channelController.synchronize { error in
+    if let error = error {
+        // Handle error
+    } else {
+        let channel = channelController.channel
+        let messages = channelController.messages
+        // Use `channel` and `messages` as you please
+    }
+}
+```
+
+To create a new Channel, we can use `ChannelController` in slightly different way:
+```swift
+let channelController = chatClient.channelController(createChannelWithId: ChannelId(type: .messaging, id: "general"))
+// Our channel is created on backend with the `synchronize` call
+channelController.synchronize { error in
+    if let error = error {
+        // Handle error
+    } else {
+        let channel = channelController.channel
+        let messages = channelController.messages
+        // Use `channel` and `messages` as you please
+    }
+}
+```
+
+Let's dive deep into these.
+
+## Channel ID
+
+In StreamChat, each Channel has a unique Channel ID, created from `type` and `id`. These will be represented in the format `type:id` in requests and backend.
+For example,
+```swift
+let channelId = ChannelId(type: .messaging, id: "general")
+```
+Encoded, this will be represented as `messaging:general`.
+For more information about channel types, please refer to [Channel Types](https://getstream.io/chat/docs/ios-swift/channel_features/?language=swift)
+
+## Different Initializers
+
+There are multiple ways of creating a `ChannelController` instance from `ChatClient`. Each way represent a unique usecase, explained below.
+
+Initializing | Use Case | Implications
+-- | -- | --
+`chatClient.channelController(for cid:)` | You already know the `ChannelId` and the Channel is already created on backend | You can directly call actions on this controller without having to call `synchronize`
```

Comment:
Seems like this table's UX could also be improved if we made the first column narrower
<img width="771" alt="Screenshot 2021-05-21 at 18 32 52" src="https://user-images.githubusercontent.com/7489425/119162845-86248100-ba63-11eb-83a1-81dc41109c33.png">

Reply:
I've tried a couple of solutions from https://stackoverflow.com/questions/36121672/set-table-column-width-via-markdown/51701842 but none seems to work with Docusaurus... Leaving it for now, to improve later